File: backend/app/schemas/signal.py
from datetime import datetime

from pydantic import BaseModel


class SignalSource(BaseModel):
    name: str
    vote: str    # bullish | bearish | neutral
    weight: float
    reason: str


class SignalResponse(BaseModel):
    symbol: str
    action: str          # buy | sell | hold
    confidence: float    # 0-100
    reasoning: list[str]
    sources: list[SignalSource]
    timeframe: str
    risk_level: str
    strategy_type: str
    generated_at: datetime

File: frontend/src/types/signal.ts
export type Action = 'buy' | 'sell' | 'hold'
export type Timeframe = 'short' | 'medium' | 'long'
export type RiskLevel = 'low' | 'medium' | 'high'
export type StrategyType = 'scalping' | 'swing' | 'long_term'

export interface SignalSource {
  name: string
  vote: 'bullish' | 'bearish' | 'neutral'
  weight: number
  reason: string
}

export interface SignalResponse {
  symbol: string
  action: Action
  confidence: number
  reasoning: string[]
  sources: SignalSource[]
  timeframe: Timeframe
  risk_level: RiskLevel
  strategy_type: StrategyType
  generated_at: string
}
